wolfcrypt/src/asn.c

```diff
@@ -8642,27 +8644,29 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
 
     #ifndef NO_SKID
         if (cert->extSubjKeyIdSet == 0 && cert->publicKey != NULL &&
-                                                        cert->pubKeySize > 0) {
+                                                         cert->pubKeySize > 0) {
             ret = CalcHashId(cert->publicKey, cert->pubKeySize,
                                                             cert->extSubjKeyId);
             if (ret != 0)
                 return ret;
         }
     #endif /* !NO_SKID */
 
-        if (cert->selfSigned) {
-            cert->maxPathLen = WOLFSSL_MAX_PATH_LEN;
-         } else {
+        if (!cert->selfSigned || (verify != NO_VERIFY && type != CA_TYPE &&
+                                                   type != TRUSTED_PEER_TYPE)) {
             cert->ca = NULL;
     #ifndef NO_SKID
             if (cert->extAuthKeyIdSet)
                 cert->ca = GetCA(cm, cert->extAuthKeyId);
-            if (cert->ca == NULL && cert->extSubjKeyIdSet \
-                                 && verify != VERIFY_OCSP) {
-                cert->ca = GetCA(cm, cert->extSubjKeyId);
```

Comment:
@SparkiDev : Did you mean to remove the lines for getting the CA based on `extSubjKeyId`? I recall this was added for a specific use case.

Reply:
We don't have a test for this.

Code added in 45703a8d72fca5b3aef1001c96775a3e28e8cf1c with comment:
handle OCSP verified certificate which is already loaded as CA

The code change doesn't make sense.
You are setting the CA of the certificate to be itself - which is only true if it is self-signed and this is not checked.

Reply:
Tesfa, what is getting the CA based on the subject key Id meant to be for?

Reply:
Check added in: de21642db113c2067cc4f88b4951cb73e05656a5

Can't see an explanation but will put it back.

Reply:
@SparkiDev : It looks like this has not yet been added back?

Reply:
Sorry! Didn't push the commit.
New push is in one commit and allows merging.